ClimsoftVer4/ClimsoftVer4/frmNewFormDaily2.vb

```diff
@@ -0,0 +1,290 @@
+
+        ucrDaiy2Navigation.PopulateControl()
+        SaveEnable()
+
+    End Sub
+
+    Private Sub cmdAssignSameValue_Click(sender As Object, e As EventArgs) Handles cmdAssignSameValue.Click
+        ucrFormDaily.SetSameValueToAllObsElements(ucrInputValue.GetValue())
+    End Sub
+
+    Private Sub btnAddNew_Click(sender As Object, e As EventArgs) Handles btnAddNew.Click
+        Dim dctSequencerFields As New Dictionary(Of String, List(Of String))
+
+        btnAddNew.Enabled = False
+        btnClear.Enabled = True
+        btnDelete.Enabled = False
+        btnUpdate.Enabled = False
+        btnCommit.Enabled = True
+
+        If chkEnableSequencer.Checked Then
+            ' temporary until we know how to get all fields from table without specifying names
+            dctSequencerFields.Add("elementId", New List(Of String)({"elementId"}))
+            ucrDaiy2Navigation.NewSequencerRecord(strSequencer:=txtSequencer.Text, dctFields:=dctSequencerFields, lstDateIncrementControls:=New List(Of ucrDataLinkCombobox)({ucrMonth}), ucrYear:=ucrYearSelector)
+        End If
+
+        ucrFormDaily.Focus()
+    End Sub
+
+    Private Sub btnCommit_Click(sender As Object, e As EventArgs) Handles btnCommit.Click
+        Try
+            If Not ValidateValues() Then
+                Exit Sub
+            End If
+
+            'Confirm if you want to continue and save data from key-entry form to database table
+            If MessageBox.Show("Do you want to continue and commit to database table?", "Save Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.Yes Then
+                ucrFormDaily.SaveRecord()
+                ucrDaiy2Navigation.GoToNewRecord()
+                SaveEnable()
+                MessageBox.Show("New record added to database table!", "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+            Else
+                MessageBox.Show("Record not Saved", "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+            End If
+        Catch ex As Exception
+            MessageBox.Show("New Record has NOT been added to database table. Error: " & ex.Message, "Save Record", MessageBoxButtons.OK, MessageBoxIcon.Error)
+        End Try
+    End Sub
+
+    Private Sub btnUpdate_Click(sender As Object, e As EventArgs) Handles btnUpdate.Click
+        Try
+            If Not ValidateValues() Then
+                Exit Sub
+            End If
+
+            If Not chkRepeatEntry.Checked Then
+                If MessageBox.Show("Are you sure you want to update this record?", "Update Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = DialogResult.Yes Then
+                    ucrFormDaily.SaveRecord()
+                    MessageBox.Show("Record updated successfully!", "Update Record", MessageBoxButtons.OK, MessageBoxIcon.Information)
+                End If
+            End If
+
+            ' When in double entry mode just skip update action. Values have been updated on entry
```

Comment:
Although values are updated on entry in the entity framework model, they are not sent to the database unless `ucrFormDaily.SaveRecord()` is called. Once the values of the key controls are changed by the sequencer to go to a new record, the current record values will also be lost from the entity framework, as this is reset in the user controls. I think this might need some further conceptualised with the new user controls/forms and entity framework model.

Reply:
@dannyparsons the implementation of double key entry has nothing to do with the design of the new key entry forms and EF. The best term to use is Repeat Entry for QC purposes. Under this implementation entry can be repeated severally if need be. It simply works as follows:

- Data is first uploaded to the 'observationinitial' table.
-Settings are done through the General Settings for which forms to be put in a repeat entry mode.
 -When a form is in repeat entry mode data entered will be compared with the equivalent record from 'observationinitial' table. If correction is necessary it will happen in the observationinitial but not in the key entry table. The original data in the key entry table will always remain unchanged.

Note:
-Functions for entries comparisons had been implemented earlier and are outside the key entry forms and EF designs.
- There are only few lines of code added in the new key entry forms for the purposes of:
      - Passing the entered value to the comparisons and updating functions.
      - Ensuring that no records are added to the key entry tables when in Repeat Entry mode

The concept is as simple as that. I have tested the implementation with the new forms and they work ok. Therefore there should be no fear that anything requires changing. Everything is already in place. 
The the few conflicts that occurred while merging may have been caused by some changes I did to remove some obvious bugs in the new forms that had nothing to do with the repeat entry implementation. Am sure when those conflicts are resolved everything will be ok

Reply:
@smachua Thanks for the clarification. I was not aware of the earlier implementation of entry comparison outside the EF model. I can now see that the repeat entry is done outside the new system of the EF model and the new controls.